rs/execution_environment/tests/execution_test.rs

```diff
@@ -2893,3 +2893,78 @@ fn no_critical_error_on_empty_data_segment() {
         "Wasm module has invalid data segment of 0 bytes at 1.",
     );
 }
+
+#[test]
+fn failed_stable_memory_grow_cost_and_time_single_canister() {
+    const NUM_WASM_PAGES: u64 = 116 * 1024 * MIB / 65_536;
+
+    let env = StateMachineBuilder::new()
+        .with_subnet_type(SubnetType::Application)
+        .build();
+
+    let canister_id = create_universal_canister_with_cycles(&env, None, INITIAL_CYCLES_BALANCE);
+
+    let timer = std::time::Instant::now();
+    let initial_balance = env.cycle_balance(canister_id);
+    let _res = env.execute_ingress(
+        canister_id,
+        "update",
+        wasm()
+            .stable64_grow(NUM_WASM_PAGES)
+            .stable64_write(0, &[42])
+            .trap()
+            .build(),
+    );
+    let elapsed_ms = timer.elapsed().as_millis();
+    let cycles_m = (initial_balance - env.cycle_balance(canister_id)) / 1000 / 1000;
```

Comment:
What is the reason for keeping track of spent cycles when they are only used to print messages when a test fails? The same applies to the test below.